sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/DataTypeName.java

```diff
@@ -5,39 +5,44 @@
 package org.lfenergy.compas.sct.commons.dto;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.NonNull;
-import lombok.Setter;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 
 @Getter
 @Setter
 @Slf4j
 @NoArgsConstructor
+@SuperBuilder
 public class DataTypeName {
     protected String name = ""; // dataName or DataAttributeName
 
+    @Builder.Default
     private List<String> structNames = new ArrayList<>(); // [.DataName[…]] or [.DAComponentName[ ….]]
 
     public DataTypeName(String dataName){
+        this();
```

Comment:
> So all of this additional features (@SuperBuilder, this(); ...) are only required by tests ? Do we really want to introduce this complexity to production code ?

To be honest, I didn't know why `this()` was required here, but without it, the field initialization of `structNames` would not work with `@Builder.Default`. I now understand why : there is an issue on lombok about this : https://github.com/projectlombok/lombok/issues/2340.

If we did not use custom constructors, we would not have this problem. I am not a fan of custom constructors, I prefer static method to create objects as recommended in the book [Effective Java](https://www.baeldung.com/java-constructors-vs-static-factory-methods#advantages-of-static-factory-methods-over-constructors). So, I think we should replace all the specific constructors by static methods, and we would not have to add `this()`. The builder pattern is a good complement here I think because there are many optional fields in the child classes (DaTypeName/DoTypeName). And yes it is very useful for testing, but not only.

> furthermore, the @SuperBuilder could be avoided by using composition over inheritance principle ...
But we would use `@Builder` anyway, no ?

Reply:
If I ckeck the production code I can see that the only way we construct `DaTypeName` is by : 
```java
public DaTypeName(String daName) {
        super(daName);
    }
```
and `DaTypeName` : 
```java
public DataTypeName(String dataName){
        this();
        if(dataName == null) return;
        String[] tokens = dataName.split("\\.");
        name = tokens[0];
        if(tokens.length > 1){
            structNames.addAll(List.of(tokens).subList(1, tokens.length));
        }
    }
```
So I repeat my question : "Do we really want to introduce this complexity to production code whereas it is only used by tests ?"

I totally agree that builder pattern could be very helpfull, it depends on the object creation.

> But we would use @Builder anyway, no ?

We could, but here it seems that a utility method for tests we be suitable

Reply:
OK. I removed Superbuilder and added methods `createDa` and `createDo` in my test class.